yt_dlp/__init__.py

```diff
@@ -1,8 +1,8 @@
 try:
-    import contextvars  # noqa: F401
+    import importlib.metadata  # noqa: F401
 except Exception:
     raise Exception(
-        f'You are using an unsupported version of Python. Only Python versions 3.7 and above are supported by yt-dlp')  # noqa: F541
+        f'You are using an unsupported version of Python. Only Python versions 3.8 and above are supported by yt-dlp')  # noqa: F541
```

Comment:
Imo we should `raise ImportError(...) from None` here. `from None` since its not caused by the failing import, and `ImportError` bc thats more aligned with what people would expect:
```py
try:
    import yt_dlp
except ImportError:
    ...  # Handle no yt_dlp
```
I also dont understand why we are not doing
```py
if sys.version < (3, 8):
    raise ImportError(...)
```

Reply:
> I also dont understand why we are not doing
> 
> ```python
> if sys.version < (3, 8):
>     raise ImportError(...)
> ```

The point of this is only to give a better error message in versions that won't ever reach `_get_system_deprecation`.

Agree with the rest